packages/beacon-node/src/api/impl/beacon/state/index.ts

```diff
@@ -296,5 +297,37 @@ export function getBeaconStateApi({
         meta: {executionOptimistic, finalized},
       };
     },
+
+    async getPendingDeposits({stateId}, context) {
+      const {state, executionOptimistic, finalized} = await getState(stateId);
+      const fork = config.getForkName(state.slot);
+
+      if (!isForkPostElectra(fork)) {
```

Comment:
We probably need to do this fork check before calling `getState()` to avoid invalid requests triggering regen by deriving fork info from state id.

I also realize other state-related endpoints also call `getState()` first, then do the request validation after. So maybe it's not that big of a deal

Reply:
> by deriving fork info from state id.

this is not easily possible as state id could be the state root, the fork check is really just a sanity check so we can safely typecast the state and give the user a proper error response

> triggering regen

I didn't add historical query support here, otherwise would have to use `getStateResponseWithRegen()` but I don't see why someone would need this information from historical states, if that's required we can enable it later on.